### ISSUE TITLE
dj.conn() has no `port` kwarg

### ISSUE THREAD

**Issue by mspacek** [datajoint/datajoint-python#563]
From what I can tell, valid kwargs for `dj.conn()` are currently:
```python
conn(host=None, user=None, password=None, init_fun=None, reset=False)
```
This is a bit of a disconnect with `dj.config`, which also has a `port` field. Currently, to programmatically set the port, I have to first overwrite the value in `dj.config`, I can't do it when setting up the connection with `dj.conn`.

It seems trivial to add a `port` kwarg to `conn`. Is it, or am I missing something?

**Comment by ixcat** [comment]
The term `host` is actually not very strict - it should accept hostname, ip, or either of these + ':port'
Just tested on a local mysql using `localhost:3306` (OK- server was running here) and `localhost:3307` (NOK - as expected, no server on this port).

**Comment by dimitri-yatsenko** [comment]
yes, I think the idea is that host just appends the port if necessary.   Is there a good reason to separate the port into a separate kwarg or should we just document this better?

**Comment by mspacek** [comment]
Sorry for the delay. Ah yes, appending the port to the host makes sense. But, it's also a bit of string wrangling, and is inconsistent with `dj.config`. It also means you can't easily specify *only* the hostname, or *only* the port, and leave the other as the default specified in `dj.config`.

Having a separate `port` kwarg would make more sense to me. I suppose for backward compatibiliy, `dj.conn` could still accept `hostname:3306` style host strings, and then raise an error if the `port` kwarg is also specified.

**Comment by dimitri-yatsenko** [comment]
good points.  Let's add `port` as a separate keyword.
